guides/code-signing/code-signing-macos.md

```diff
@@ -62,44 +74,98 @@ To enable code signing on macOS, ensure that `packagerConfig.osxSign` exists in
+      ],
+      identity: 'Apple Development',
+      platform: 'mas',
+      type: 'development',
+      provisioningProfile: 'development.provisionprofile',
+      optionsForFile: (filePath) => {
+        const entitlements = filePath.includes('.app/') ? 'entitlements.child.plist' : 'entitlements.plist';
+        return {
+          hardenedRuntime: false,
+          entitlements
+        }
+      }
+    }
   }
 };
 ```
 {% endcode %}
 
-The `osxSign` config comes with defaults that work out of the box in most cases, so we recommend you start with an empty configuration object.
+```binaries```: if your electron app calls any binaries, they need to be listed here so that they can be signed.
 
-For a full list of configuration options, see the [`OsxSignOptions`](https://js.electronforge.io/modules/\_electron\_forge\_shared\_types.InternalOptions.html#OsxSignOptions) type in the Forge API docs. For more detailed information on how to configure these options, see the [`@electron/osx-sign` documentation](https://github.com/electron/osx-sign).
+```identity```: the name of the certificate.
 
-#### Customizing entitlements
+- App store development: Apple Development
+- App store distribution: Apple Distribution: FirstName LastName (TEAMID)
+- Outside distribution: Developer ID Application: FirstName LastName (TEAMID)
 
-A common use case for modifying the default `osxSign` configuration is to customize its entitlements. In macOS, **entitlements** are privileges that grant apps certain capabilities (e.g. access to the camera, microphone, or USB devices). These are stored within the code signature in an app's executable file.
+```platform```: for the app store it is ```mas``` and for outside the app store it is ```darwin```
 
-By default, the `@electron/osx-sign` tool comes with a set of entitlements that should work on both MAS or direct distribution targets. See the complete set of default entitlement files [on GitHub](https://github.com/electron/osx-sign/tree/main/entitlements).
+```provisioningProfile```: the appropriate provisioning profile, as mentioned earlier.
 
-{% code title="forge.config.js" %}
-```javascript
-module.exports = {
-  // ...
-  packagerConfig: {
-    // ...
-    osxSign: {
-      optionsForFile: (filePath) => {
-        // Here, we keep it simple and return a single entitlements.plist file.
-        // You can use this callback to map different sets of entitlements
-        // to specific files in your packaged app.
-        return {
-          entitlements: 'path/to/entitlements.plist'
-        };
-      }
-    }
-  }
-  // ...
-};
+```optionsForFile```: for distribution outside of the app store, you may be able to rely on the defaults if you app doesn't need any extra entitlements. For the app store, you will definitely need to provide this.
+
+You need to add logic to determine which set of entitlements to use. If you specify more entitlements then your app uses, it will probably be rejected by the review process.
```

Comment:
```diff
- You need to add logic to determine which set of entitlements to use. If you specify more entitlements then your app uses, it will probably be rejected by the review process.
+ Your list of entitlements should contain only the ones your application needs. Adding extraneous entitlements could be cause for rejection during the review process.
```